src/vips/Image_methods.lua

```diff
@@ -521,6 +530,32 @@ function Image_method:filename()
     return self:get("filename")
 end
 
+-- misc query functions
+
+function Image_method:is_sequential()
+    return call_for_bool(self, vips_lib.vips_image_is_sequential)
+end
+
+function Image_method:isMSBfirst()
+    return call_for_bool(self, vips_lib.vips_image_isMSBfirst)
+end
+
+function Image_method:isfile()
+    return call_for_bool(self, vips_lib.vips_image_isfile)
+end
+
+function Image_method:ispartial()
+    return call_for_bool(self, vips_lib.vips_image_ispartial)
+end
+
+function Image_method:hasalpha()
+    return call_for_bool(self, vips_lib.vips_image_hasalpha)
+end
```

Comment:
I agree `hasalpha` is needed, but could you explain the rationale for the other new methods? None of the other language bindings have these, and I'm not sure they are necessary.

They are used in libvips to implement operations, but I don't think you'd need them if you are using operations.

Reply:
When I was implementing `hasalpha` support, I saw that the other functions exist, and since adding them too was trivial, I figured there'd be no harm in it. I've got no need for them besides `hasalpha`, and maybe `is_sequential`, which, from what I understand, could occasionally be useful for checking whether non-sequential operations can be used on an image.